add_pricelist_price/models/sale_order.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    # computational field for book price
+    book_price = fields.Float(
+        string="Book Price", 
+        compute="_compute_book_price", 
+        help="Price without any discount and taxes",
+    )
+
+    @api.depends('product_id', 'product_uom_qty', 'order_id.pricelist_id')
+    def _compute_book_price(self):
+        # function for computing book price depends on product, quantity and pricelist
+        for line in self:
+            if line.product_id and line.order_id:
+                book_price = line.order_id.pricelist_id._get_product_price(
+                    product=line.product_id, 
+                    quantity=line.product_uom_qty,
+                    uom=line.product_uom, # unit of measure
+                )
+                line.book_price = book_price or line.product_id.lst_price
+            else:
+                line.book_price = line.product_id.lst_price if line.product_id else 0.0
```

Comment:
```suggestion
                line.book_price = line.product_id.lst_price
```
`lst_price` is Float field
```py
In [2]: self.env['product.product'].lst_price
Out[2]: 0.0
```
in any case you try to perform on a empty record set it. ORM will return `0.0`

Reply:
True 🙂, we don't need that if statement.